src/renderer/components/ResponseCodeError/ResponseCodeError.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from "react";
+import {useState, useEffect} from "react";
+import {PieChart, Pie, Tooltip, Cell,} from "recharts";
+import {IntervalEnum} from "../BalanceGraph/BalanceGraph";
+
+export interface IResponseCodeErrorProps {
+
+    getData: (interval: IntervalEnum) => Promise<Array<object>>;
+}
+
+const COLORS = ["#09BC8A", "#EDFF86", "#EA526F"];
+
+export const ResponseCodeError: React.FunctionComponent<IResponseCodeErrorProps> = 
+(props: IResponseCodeErrorProps) => {
+    const [data, setData] = useState<Array<object>>([]);
+    const [intervalOption, setIntervalOption] = useState<IntervalEnum>(IntervalEnum.DAY);
+    const [refreshIntervalId, setRefreshIntervalId] = useState<number>(0);
+
+    const handleOptionClick = (IntervalValue: IntervalEnum): void=>{
+        setIntervalOption(IntervalValue);
+    };
+
+    const renderGraphIntervalOption = (IntervalValue: IntervalEnum): string => {
+        let selector = "";
+        intervalOption===IntervalValue ? selector="selected" : null;
+        return `graph-option ${selector}`;
+    };
+
+    const awaitData = async (): Promise<void>=>{
```

Comment:
you aren't really waiting data. If you wan't something to happen after this you have to do
```javascript
const dataValueArray = await props.getData(intervalOption);
setData(dataValueArray);
```

Reply:
@irubido I've gave you the real snippet, your code won't do much :smile: